src/electron.js

```diff
@@ -29,30 +25,53 @@ const Environment = Object.freeze({
 
 const args = process.argv.slice(2)
 
-let appEnvironment, localhostPort
-// NOTE: This args will only apply if running electron itself locally
-// Production builds with installers (staging and prod) cannot receive args like this.
+let appEnvironment, localhostPort, scheme
+
+// Try getting the env from the program args if available.
+// Otherwise, try getting env from the electronConfig file.
+let env = null
 if (args.length > 0) {
-  switch (args[0]) {
-    case 'localhost':
-      appEnvironment = Environment.LOCALHOST
-      localhostPort = args[1] || '3000'
-      break
-    case 'staging':
-      appEnvironment = Environment.STAGING
-      break
-    case 'production':
-      appEnvironment = Environment.PRODUCTION
-      break
-    default:
-      appEnvironment = ''
-      break
+  env = args[0]
+}
+if (!env) {
+  const configFile = path.resolve(app.getAppPath(), 'electronConfig.json')
+  try {
+    // Needs to be readFileSync because async js here will take
+    // too long, leading the scheme to be decided after the app has
+    // registered the scheme.
+    const data = fs.readFileSync(configFile)
+    const config = JSON.parse(data)
+    env = config.env
+  } catch (e) {
+    console.error('No config file found', e)
```

Comment:
Nit: Maybe specify that the error could have occurred because something was wrong with the config file content

Reply:
👌 yes!